### ISSUE TITLE
Remove Student from course causes fetch to fail

### ISSUE THREAD

**Issue by bjcoleman** [git-keeper/git-keeper#136]
Jeff reports that he used `gkeep modify` to drop a student and then `gkeep fetch ...` to get results from an assignment.  The fetch failed with (student id omitted):

```
Traceback (most recent call last):
  File "/usr/local/bin/gkeep", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.7/site-packages/gkeepclient/gkeep.py", line 376, in main
    take_action(parsed_args)
  File "/usr/local/lib/python3.7/site-packages/gkeepclient/gkeep.py", line 415, in take_action
    dest_path)
  File "/usr/local/lib/python3.7/site-packages/gkeepclient/client_function_decorators.py", line 51, in wrapper
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/gkeepclient/client_function_decorators.py", line 77, in wrapper
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/gkeepclient/client_function_decorators.py", line 107, in wrapper
    return func(class_name, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/gkeepclient/fetch_submissions.py", line 361, in fetch_submissions
    destination_path, info)
  File "/usr/local/lib/python3.7/site-packages/gkeepclient/fetch_submissions.py", line 261, in fetch_assignment_submissions
    remote_head_hash = info.student_assignment_hash(class_name, assignment_name, username)
  File "/usr/local/lib/python3.7/site-packages/gkeepcore/faculty_class_info.py", line 278, in student_assignment_hash
    return assignment_info['students_repos'][username]['hash']
KeyError: '(omitted)'
```

**Comment by bjcoleman** [comment]
I have not verified.  Posting because he says he had the same error last semester - and I don't want this to get lost again.

I assume it's not user error... :)

**Comment by coderforlife** [comment]
Workaround is to restart gkeepd on server.

One thing to note is that the assignment was uploaded/published AFTER the student was dropped. So the assignment was never given to them since they had always been dropped. Fetching still tries to fetch their repo.

**Comment by coderforlife** [comment]
I believe this is a minimal non-working example:

```
gkeep modify classname classname.csv # drops a student
gkeep upload classname assignment
gkeep publish classname assignment
gkeep fetch classname assignment
```

**Comment by coderforlife** [comment]
As a note, this error has now shown up with an assignment published only about 1.5 weeks ago, the student had been dropped from the course for about 7 weeks (with multiple gkeepd restarts in between) and a fetch on a different assignment only a few days prior had no problems.

Similar (if not same) trace:

```
Traceback (most recent call last):
  File "/usr/local/bin/gkeep", line 11, in <module>
    sys.exit(main())
  File "/usr/local/lib/python3.7/site-packages/gkeepclient/gkeep.py", line 376, in main
    take_action(parsed_args)
  File "/usr/local/lib/python3.7/site-packages/gkeepclient/gkeep.py", line 415, in take_action
    dest_path)
  File "/usr/local/lib/python3.7/site-packages/gkeepclient/client_function_decorators.py", line 51, in wrapper
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/gkeepclient/client_function_decorators.py", line 77, in wrapper
    return func(*args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/gkeepclient/client_function_decorators.py", line 107, in wrapper
    return func(class_name, *args, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/gkeepclient/fetch_submissions.py", line 361, in fetch_submissions
    destination_path, info)
  File "/usr/local/lib/python3.7/site-packages/gkeepclient/fetch_submissions.py", line 261, in fetch_assignment_submissions
    remote_head_hash = info.student_assignment_hash(class_name, assignment_name, username)
  File "/usr/local/lib/python3.7/site-packages/gkeepcore/faculty_class_info.py", line 278, in student_assignment_hash
    return assignment_info['students_repos'][username]['hash']
KeyError: '(redacted)'
```
